packages/console/src/pages/ApplicationDetails/SamlApplicationDetailsContent/Settings.tsx

```diff
@@ -186,14 +234,49 @@ function Settings({ data, mutateApplication, isDeleted }: Props) {
             </>
           )}
           <FormField title="application_details.saml_idp_certificates.title">
-            <Table
-              hasBorder
-              isRowHoverEffectDisabled
-              rowIndexKey="id"
-              isLoading={!secrets.data && !secrets.error}
-              errorMessage={secrets.error?.body?.message ?? secrets.error?.message}
-              rowGroups={[{ key: 'application_secrets', data: secrets.data ?? [] }]}
-              columns={secretTableColumns}
+            {secrets.data?.length === 0 ? (
+              <>
+                <div className={styles.empty}>{t('application_details.secrets.empty')}</div>
+                <Button
+                  icon={<Plus />}
+                  title="application_details.secrets.create_new_secret"
+                  onClick={() => {
+                    setShowCreateSecretModal(true);
+                  }}
+                />
+              </>
+            ) : (
```

Comment:
Consider using the loading skeleton here, when the `secrets.data` is not completed loaded

Reply:
Loading state added.